### PR TITLE
Support for Linux multitouch overlays and monitors

### PR DESCRIPTION
**Pull request by snibbe**
Hi Olivia, I extended this class to support multitouch overlays and monitors on Linux! I also updated and tested the code on Windows with the latest OF12. Can you accept this pull request so everyone can use the extend class on Linux? I can post about it once merged on some of the OF threads where people have asked for this functionality. If you have time, there are some funky aspects to the Windows examples, perhaps due to the latest Windows and OF, but things work well enough for someone who wants to integrate multitouch into their own app to do so.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -2,22 +2,29 @@
...
 - Download from GitHub
 - Unzip the folder and rename it from `ofxMultitouch-master` to `ofxMultitouch`
 - Place it in your `openFrameworks/addons` folder.
+- You can also clone the repo directly into your addons directory: `cd <OF>/addons` and then `git clone <repository URL>`
 
 **Running examples:**
 - Import the example into the project generator
 - Create a new project
 - Open the project file in your IDE
+- On Linux, examples must be run as root, e.g. `sudo example-touchCircles`. The RunDebug and RunRelease targets for the example files have been modified to do this automatically, but you will needto enter the root password when you run them.
```
Comment (olivialynn) [inline README.md]:
```suggestion
- On Linux, examples must be run as root, e.g. `sudo example-touchCircles`. The RunDebug and RunRelease targets for the example files have been modified to do this automatically, but you will need to enter the root password when you run them.
```

File: README.md [hunk 2]
```diff
@@ -2,22 +2,29 @@
...
 
-*(Made and tested on Windows 10, openFrameworks 9.8 and 10.1, and Visual Studio 2017 - feel free to reach out if you run into problems in other environments)*
+*(Windows version made and tested on Windows 10, openFrameworks 9.8 and 10.1, and Visual Studio 2017 - feel free to reach out if you run into problems in other environments)*
+*Linux version made and tested on Linux Mint, openFramworks 12*
```
Comment (olivialynn) [inline README.md]:
```suggestion
*Linux version made and tested on Linux Mint, openFrameworks 12*
```

File: example-imageGestures/addons.make [hunk 3]
```diff
@@ -1 +1,2 @@
 ofxMultitouch
+ofxMultitouch
```
Comment (olivialynn) [inline example-imageGestures/addons.make:2]: In all honestly I don’t remember anything about the addon.make file, so I’m working off the [two sentences in the docs](https://openframeworks.cc/learning/01_basics/how_to_add_addon_to_project/#:~:text=a%20file%20called-,addons.make,-.%20It%20should%20be) that covers it - is the duplication here and in the other two addons.make files intentional?

**Review by olivialynn — APPROVED** [review]
Wow, incredible! I wrote this as an undergrad completely new to oF (and C++, iirc), so my condolences for any spaghetti you’ve read through.

Small disclaimer that I no longer have access to the multitouch screen I wrote this for, so I can’t test any of this out on my own for the time being.

That said, this looks great and I’m excited to have Linux support for this project now!

File: example-imageGestures/addons.make (hunk 3 above)
Comment (snibbe) [reply example-imageGestures/addons.make:2]: The additional line isn't required. ProjectGenerator starts adding multiple copies of add-ons as it's run multiple times, so this second line can be removed.

**Comment by snibbe** [comment]
Thanks so much! I tested on Windows and the basics work well enough, though there is some funkiness with all the examples except touchCircles. If I have time I may clean up some of the other Windows code later. And I'm going to add in instructions for making touch devices readable on Linux by non-root users once I figure that out. One thing you could do is enable "issues" for this repo so we can add in bug reports and feature requests. Glad to be collaborating with you to extend this useful class!

**Comment by snibbe** [comment]
Let me know if you want me to make any other changes before accepting this. I'll add in instructions for making the touch screens user-writable on Linux later, once I figure it out (issue #3).

**Comment by olivialynn** [comment]
> Let me know if you want me to make any other changes before accepting this. I'll add in instructions for making the touch screens user-writable on Linux later, once I figure it out (issue #3).

Apologies for the delay - feel free to merge whenever!

**Comment by snibbe** [comment]
I think you may need to merge, or add me as a "contributor" to this project?

**Comment by snibbe** [comment]
> > Let me know if you want me to make any other changes before accepting this. I'll add in instructions for making the touch screens user-writable on Linux later, once I figure it out (issue #3).
> 
> Apologies for the delay - feel free to merge whenever!

I think you may need to merge, or add me as a "contributor" to this project?

**Pull request merged**